### PR TITLE
Remove RIDS in CompareMsftToSbAssemblyVersions test

### LINKED ISSUE #4493: [Main] All SDK Diff Tests Failing
As of today, all three SDK diff tests are failing: https://dev.azure.com/dnceng/internal/_build/results?buildId=2490356&view=results (internal Microsoft link)

### PR DESCRIPTION
**Pull request by ellahathaway**
Closes https://github.com/dotnet/source-build/issues/4493

When testing the changes in https://github.com/dotnet/sdk/pull/41929, the `CompareMsftToSbAssemblyVersions` was continuing to fail with the following error message:

```bash
Unable to find matching file for 'packs/runtime.centos.9-x64.Microsoft.DotNet.ILCompiler/x.y.z/sdk/System.Private.StackTraceMetadata.dll' in '/tmp/t3vand3s.y0t/msft'
```

This file should fall under the following exclusion that exists in the exclusions file: `./packs/runtime.banana-rid.Microsoft.DotNet.ILCompiler/|sb`. Upon investigation, I discovered that the issue was that unlike the `CompareMsftToSbFileList` test, the `CompareMsftToSbAssemblyVersions` test does not remove the RID from the file.

The fix here was to 1) Remove the RID from the SB file, and 2) replace the placeholder rids with matching patterns when attempting to file match to the Msft SDK.

- [Failing test without changes](https://dev.azure.com/dnceng/internal/_build/results?buildId=2490565&view=logs&j=c2a3fba4-babc-5882-e82a-8e0dc9edd6ec&t=c2a95e7e-b619-54b2-c512-fcefc49a38ba) (internal Microsoft link)
- [Successful test with changes](https://dev.azure.com/dnceng/internal/_build/results?buildId=2490617&view=logs&j=c2a3fba4-babc-5882-e82a-8e0dc9edd6ec&t=c2a95e7e-b619-54b2-c512-fcefc49a38ba) (internal Microsoft link)

We should merge this before merging in https://github.com/dotnet/sdk/pull/41929

### CODE REVIEW

File: src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BaselineHelper.cs [hunk 1]
```diff
@@ -119,14 +123,16 @@ public static string RemoveVersions(string source)
...
+        /// <see cref="RemoveVersions(string)"/> and <see cref="RemoveRids(string, bool)"/>
         /// </summary>
         public static Matcher GetFileMatcherFromPath(string path)
         {
             path = path
                 .Replace(SemanticVersionPlaceholder, SemanticVersionPlaceholderMatchingPattern)
-                .Replace(NonSemanticVersionPlaceholder, NonSemanticVersionPlaceholderMatchingPattern);
+                .Replace(NonSemanticVersionPlaceholder, NonSemanticVersionPlaceholderMatchingPattern)
+                .Replace(TargetRidPlaceholder, TargetRidPlaceholderMatchingPattern)
+                .Replace(PortableRidPlaceholder, PortableRidPlaceholderMatchingPattern);
```
Comment (MichaelSimons) [inline src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BaselineHelper.cs:135]: Given the Target and Portable rid placeholders are the same, how would the portable replace ever be needed?

Comment (ellahathaway) [reply src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BaselineHelper.cs:135]: The Target rid placeholder is `banana-rid` and the Portable rid placeholder is `portable-rid`. They're matching pattern placeholders are both `*-*` though.

Comment (MichaelSimons) [reply src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BaselineHelper.cs:135]: I transposed the arguments in my mind.  I see that it is doing now.

**Review by MichaelSimons — APPROVED** [review]

**Comment by ellahathaway** [comment]
@dotnet/source-build-internal - the failures are unrelated to these changes. Can someone please bypass the checks for this PR? These changes are necessary for a green SDK diff pipeline run. TIA

**Pull request merged**